File: VB/T589084_VB/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc

Imports System.Web.Mvc
Imports System.Web.Script.Serialization

Namespace Controllers
    Public Class HomeController
        Inherits Controller

        ' GET: Home
        Function Index() As ActionResult
            Return View()
        End Function

        Public Function RichEditPartial() As ActionResult
            Return PartialView("_RichEditPartial")
        End Function

        Public Function GetJsonData() As String
            Dim jsSerializer As JavaScriptSerializer = New JavaScriptSerializer()
            Dim parentRow As List(Of Dictionary(Of String, Object)) = New List(Of Dictionary(Of String, Object))()
            Dim childRow As Dictionary(Of String, Object)
            Dim table As DataTable = MyModel.GetDT2()
            For Each row As DataRow In table.Rows
                childRow = New Dictionary(Of String, Object)()
                For Each col As DataColumn In table.Columns
                    childRow.Add(col.ColumnName, row(col))
                Next

                parentRow.Add(childRow)
            Next

            Return jsSerializer.Serialize(parentRow)
        End Function

        Public Function _RichEditPartial() As ActionResult
            Return PartialView("_RichEditPartial")
        End Function
    End Class
End Namespace

File: CS/T589084/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Web.Script.Serialization;
using T589084.Models;

namespace T589084.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();    
        }
        public ActionResult RichEditPartial()
        {
            return PartialView("_RichEditPartial");
        }
        public string GetJsonData()
        {
            JavaScriptSerializer jsSerializer = new JavaScriptSerializer();
            List<Dictionary<string, object>> parentRow = new List<Dictionary<string, object>>();
            Dictionary<string, object> childRow;
            DataTable table = MyModel.GetDT2();
            foreach (DataRow row in table.Rows)
            {
                childRow = new Dictionary<string, object>();
                foreach (DataColumn col in table.Columns)
                {
                    childRow.Add(col.ColumnName, row[col]);
                }
                parentRow.Add(childRow);
            }
            return jsSerializer.Serialize(parentRow);
        }
    }
}